tools/mc-java-protoc/src/main/java/io/spine/tools/mc/java/protoc/message/ImplementByPattern.java

```diff
@@ -42,13 +46,18 @@
  */
 final class ImplementByPattern extends ImplementInterface {
 
-    private final FilePatternMatcher matcher;
+    private final Predicate<MessageType> matcher;
+
+    ImplementByPattern(JavaClassName config, FilePattern pattern) {
+        super(config);
+        checkNotDefaultArg(pattern);
+        this.matcher = new FilePatternMatcher(pattern);
+    }
 
-    ImplementByPattern(ConfigByPattern config) {
-        super(config.getValue());
-        FilePattern filePattern = config.getPattern();
-        checkNotDefaultArg(filePattern);
-        this.matcher = new FilePatternMatcher(filePattern);
+    ImplementByPattern(JavaClassName config, Pattern pattern) {
```

Comment:
To a naked eye, this ctor seems odd, as the class-level description only mentions `FilePattern`s, not `Pattern`s.

Please consider documenting both ctors.